src/hooks/use-reduced-motion.ts

```
@@ -0,0 +1,26 @@
+ * in static exports — it captures the SSR value (null) via useState
+ * and never updates it on the client.
+ *
+ * This hook defaults to false (animations enabled) during SSR, then
+ * reads the actual preference via useEffect after mount.
+ */
+export function useReducedMotion(): boolean {
+  const [prefersReduced, setPrefersReduced] = useState(false);
+
+  useEffect(() => {
+    const mql = window.matchMedia('(prefers-reduced-motion: reduce)');
+    setPrefersReduced(mql.matches);
```

ESLint

[error] Error: Calling setState synchronously within an effect can trigger cascading renders

Effects are intended to synchronize state between React and external systems such as manually updating the DOM, state management libraries, or other platform APIs. In general, the body of an effect should do one or both of the following:
* Update external systems with the latest state from React.
* Subscribe for updates from some external system, calling setState in a callback function when external state changes.

Calling setState synchronously within an effect body causes cascading renders that can hurt performance, and is not recommended. (https://react.dev/learn/you-might-not-need-an-effect).

/home/jailuser/git/src/hooks/use-reduced-motion.ts:18:5
  16 |   useEffect(() => {
  17 |     const mql = window.matchMedia('(prefers-reduced-motion: reduce)');
> 18 |     setPrefersReduced(mql.matches);
     |     ^^^^^^^^^^^^^^^^^ Avoid calling setState() directly within an effect
  19 |
  20 |     const handler = (e: MediaQueryListEvent) => setPrefersReduced(e.matches);
  21 |     mql.addEventListener('change', handler);

(react-hooks/set-state-in-effect)